app/api/structured_extract.py

```
@@ -0,0 +1,180 @@
+                except Exception as e:
+                     raise HTTPException(status_code=500, detail=f"Error retrieving file {key} from storage.")
+
+                if is_pdf:
+                    try:
+                        with open(temp_file, "rb") as f:
+                            header = f.read(5)
+                        if header != b"%PDF-":
+                            raise HTTPException(status_code=400, detail="Downloaded file is not a valid PDF.")
+                    except HTTPException:
+                        raise
+                    except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)